kernel/frame_allocator/src/lib.rs

```diff
@@ -944,10 +944,9 @@ pub fn allocate_frames_deferred(
             // If we failed to allocate the requested frames from the general list,
             // we can add a new reserved region containing them,
             // but ONLY if those frames are *NOT* in the general-purpose region.
-            else if {
-                let g = GENERAL_REGIONS.lock();  
-                !frame_is_in_list(&g, &start_frame) && !frame_is_in_list(&g, &end_frame)
-            } {
+            else if !frame_is_in_list(&GENERAL_REGIONS.lock(), &start_frame) 
+                    && !frame_is_in_list(&GENERAL_REGIONS.lock(), &end_frame) 
```

Comment:
Isn't this changing the semantics? Before, we locked once and passed a reference to that lock twice, now we lock twice. Might not matter, but someone else should review this one again.